elasticdl/python/worker/task_data_service.py

```diff
@@ -158,6 +160,11 @@ def get_dataset(self):
                 for _ in self.data_reader.read_records(task):
                     break
                 self._has_warmed_up = True
+
+            t = threading.Thread(target=self._get_records_fn)
+            t.daemon = True
+            t.start()
```

Comment:
When does the thread end?

Reply:
Done. It ends after the  training dataset is deleted. Please refer to https://github.com/QiJune/elasticdl/blob/37b39c08f0b929a64ace07c413e13d3c652765d7/elasticdl/python/worker/worker.py#L993-L994